src/main/java/practice/StreamPractice.java

```diff
@@ -1,80 +1,65 @@
 package practice;
 
-import java.util.Collections;
+import java.util.Arrays;
 import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
 import model.Candidate;
+import model.Cat;
 import model.Person;
 
 public class StreamPractice {
-    /**
-     * Given list of strings where each element contains 1+ numbers:
-     * input = {"5,30,100", "0,22,7", ...}
-     * return min integer value. One more thing - we're interested in even numbers.
-     * If there is no needed data throw RuntimeException with message
-     * "Can't get min value from list: < Here is our input 'numbers' >"
-     */
+    private static final String SEPARATOR = ",";
+    private static final int DIVIDER = 2;
+
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        return numbers.stream()
+                .flatMap(numberString -> Arrays.stream(numberString.split(SEPARATOR)))
+                .mapToInt(Integer::parseInt)
+                .filter(number -> number % DIVIDER == 0)
+                .min()
+                .orElseThrow(() ->
+                        new NoSuchElementException("Can't get min value from list: " + numbers));
     }
 
-    /**
-     * Given a List of Integer numbers,
-     * return the average of all odd numbers from the list or throw NoSuchElementException.
-     * But before that subtract 1 from each element on an odd position (having the odd index).
-     */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+                .map(i -> i % DIVIDER == 0 ? numbers.get(i) : numbers.get(i) - 1)
+                .filter(number -> number % DIVIDER != 0)
+                .average().orElseThrow(() ->
+                        new NoSuchElementException("Can't get avarage from list: " + numbers));
```

Comment:
```suggestion
                .average().
                .getAsDouble();
```
Please refer to common mistakes